src/pages/Login.js

```diff
@@ -0,0 +1,215 @@
+    if (res.statusText === "OK") {
+      localStorage.setItem("isAuth", true);
+      localStorage.setItem("token", res.data.encodedToken);
+      localStorage.setItem("userData", JSON.stringify(res.data.foundUser));
+      notificationHandler({
+        type: "success",
+        content: "Successfully Logged In",
+      });
+    } else {
+      notificationHandler({
+        type: "warning",
+        content: "Your Password/Email Id Is Wrong, Try Again",
+      });
+    }
+  } catch (error) {
+    notificationHandler({
+      type: "warning",
+      content: "Server Is Not Responding, Try Again Later",
+    });
+    console.error(error);
+  }
+};
+
+const LoginPage = ({ closeModal }) => {
+  const [action, setAction] = useState("login");
+  const navigate = useNavigate();
+  const { dispatch, state } = useGlobalContext();
+
+  useEffect(() => {
+    if (state.user.isAuth) closeModal();
+  }, [state]);
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+    let data = {};
+    for (let x of formData) {
+      data[x[0]] = x[1];
+    }
+
+    LoginHandler(data).then((res) => {
+      setTimeout(() => {
+        dispatch({ for: "USER", type: "UPDATE" });
+        closeModal();
+        navigate(0);
+      }, 2000);
+    });
+  };
+
+  return (
+    <div className="section loginPage">
+      <div className="container">
+        {action === "login" ? (
+          <div className="login_comp flex flex-col p2">
+            <h2 className="head2">Login</h2>
+            <form className="login_form flex flex-col" onSubmit={submitHandler}>
+              <label htmlFor="username">Uername</label>
+              <div className="input">
+                <input type="text" id="username" name="email" required />
+              </div>
+              <label htmlFor="password">Password</label>
+              <div className="input">
+                <input type="password" id="password" name="password" required />
+              </div>
+              <label className="checkbox">
+                <input type="checkbox" />
+                Remember Me
+              </label>
+              <button className="btn" type="submit">
+                Login
+              </button>
+              <button
+                className="btn"
+                type="button"
+                onClick={() => setAction("signup")}
+              >
+                Create An Account {">"}
+              </button>
+              <button
+                className="btn"
+                type="button"
+                onClick={() => {
+                  LoginHandler({
+                    email: "hell@gmail.com",
+                    password: "12345678",
+                  }).then((res) => {
+                    setTimeout(() => {
+                      dispatch({ for: "USER", type: "UPDATE" });
+                      closeModal();
+                      navigate(0);
+                    }, 2000);
```

Comment:
somewhere you are using async method, somewhere you are using then method. Make a uniformity in the codes.

Reply:
yep you are right will do that